# Bonus: Temperature Analysis I

In [1]:
import pandas as pd
from datetime import datetime as dt

In [2]:
# "tobs" is "temperature observations"
df = pd.read_csv('Resources/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [3]:
# Convert the date column format from string to datetime
df.date = pd.to_datetime(df.date, infer_datetime_format=True)


In [4]:
# Set the date column as the DataFrame index
df = df.set_index(df['date'])
df.head()


station       date  prcp  tobs
date                                          
2010-01-01  USC00519397 2010-01-01  0.08    65
2010-01-02  USC00519397 2010-01-02  0.00    63
2010-01-03  USC00519397 2010-01-03  0.00    74
2010-01-04  USC00519397 2010-01-04  0.00    76
2010-01-06  USC00519397 2010-01-06   NaN    73

In [5]:
# Drop the date column
df = df.drop(columns='date')
df.head()


station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

### Compare June and December data across all years 

In [6]:
from scipy import stats

In [7]:
# Filter data for desired months
june_df = df[df.index.month == 6]
june_df


station  prcp  tobs
date                               
2010-06-01  USC00519397  0.00    78
2010-06-02  USC00519397  0.01    76
2010-06-03  USC00519397  0.00    78
2010-06-04  USC00519397  0.00    76
2010-06-05  USC00519397  0.00    77
...                 ...   ...   ...
2017-06-26  USC00516128  0.02    79
2017-06-27  USC00516128  0.10    74
2017-06-28  USC00516128  0.02    74
2017-06-29  USC00516128  0.04    76
2017-06-30  USC00516128  0.20    75

[1700 rows x 3 columns]

In [8]:
# Identify the average temperature for June
june_mean_df=june_df["tobs"].mean()
june_mean_df


74.94411764705882

In [9]:
# Identify the average temperature for December
dec_df = df[df.index.month == 12]
dec_df


station  prcp  tobs
date                               
2010-12-01  USC00519397  0.04    76
2010-12-03  USC00519397  0.00    74
2010-12-04  USC00519397  0.00    74
2010-12-06  USC00519397  0.00    64
2010-12-07  USC00519397  0.00    64
...                 ...   ...   ...
2016-12-27  USC00516128  0.14    71
2016-12-28  USC00516128  0.14    71
2016-12-29  USC00516128  1.03    69
2016-12-30  USC00516128  2.37    65
2016-12-31  USC00516128  0.90    65

[1517 rows x 3 columns]

In [10]:
# Create collections of temperature data
dec_mean_df=dec_df["tobs"].mean()
dec_mean_df


71.04152933421226

In [11]:
# Run paired t-test
june_temp=june_df.tobs
june_temp


date
2010-06-01    78
2010-06-02    76
2010-06-03    78
2010-06-04    76
2010-06-05    77
              ..
2017-06-26    79
2017-06-27    74
2017-06-28    74
2017-06-29    76
2017-06-30    75
Name: tobs, Length: 1700, dtype: int64

In [12]:
dec_temp=dec_df.tobs
dec_temp


date
2010-12-01    76
2010-12-03    74
2010-12-04    74
2010-12-06    64
2010-12-07    64
              ..
2016-12-27    71
2016-12-28    71
2016-12-29    69
2016-12-30    65
2016-12-31    65
Name: tobs, Length: 1517, dtype: int64

In [13]:
stats.ttest_ind(june_temp,dec_temp)


Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

The average temperature difference between June and December is only 3.9 °F. The t-test shows a low p-value, meaning the value is statistically different. So, even though there is a statistical difference in the temperatures between June and December, the difference is only 3.9 °F. Send me to Hawaii anytime of the year.